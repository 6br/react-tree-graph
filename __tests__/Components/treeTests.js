import React from 'react';
import { shallow } from 'enzyme';

import Link from '../../Components/link';
import Node from '../../Components/node';
import Tree from '../../Components/tree';

const data = {
	name: 'Colour',
	childs: [{
		name: 'Black'
	}]
};

describe('<Tree>', () => {
	test('renders correctly', () => {
		const props = {
			data: data,
			// Must not use children prop due to special meaning in react causing shallow renderer errors
			getChildren: d => d.childs,
			height: 100,
			width: 200,
		};

		const tree = shallow(<Tree {...props}/>);

		expect(tree).toMatchSnapshot();
	});
});