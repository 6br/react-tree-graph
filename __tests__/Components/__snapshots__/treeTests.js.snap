// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tree> renders correctly 1`] = `
<Animated
  animated={false}
  duration={500}
  easing={[Function]}
  height={100}
  keyProp="name"
  labelProp="name"
  linkClass="link"
  links={
    Array [
      Object {
        "source": Node {
          "children": Array [
            Node {
              "data": Object {
                "name": "Black",
              },
              "depth": 1,
              "height": 0,
              "parent": [Circular],
              "x": 40,
              "y": 40,
            },
          ],
          "data": Object {
            "childs": Array [
              Object {
                "name": "Black",
              },
            ],
            "name": "Colour",
          },
          "depth": 0,
          "height": 1,
          "parent": null,
          "x": 40,
          "y": 10,
        },
        "target": Node {
          "data": Object {
            "name": "Black",
          },
          "depth": 1,
          "height": 0,
          "parent": Node {
            "children": Array [
              [Circular],
            ],
            "data": Object {
              "childs": Array [
                Object {
                  "name": "Black",
                },
              ],
              "name": "Colour",
            },
            "depth": 0,
            "height": 1,
            "parent": null,
            "x": 40,
            "y": 10,
          },
          "x": 40,
          "y": 40,
        },
      },
    ]
  }
  nodeClass="node"
  nodeOffset={3.5}
  nodeRadius={5}
  nodes={
    Array [
      Node {
        "children": Array [
          Node {
            "data": Object {
              "name": "Black",
            },
            "depth": 1,
            "height": 0,
            "parent": [Circular],
            "x": 40,
            "y": 40,
          },
        ],
        "data": Object {
          "childs": Array [
            Object {
              "name": "Black",
            },
          ],
          "name": "Colour",
        },
        "depth": 0,
        "height": 1,
        "parent": null,
        "x": 40,
        "y": 10,
      },
      Node {
        "data": Object {
          "name": "Black",
        },
        "depth": 1,
        "height": 0,
        "parent": Node {
          "children": Array [
            [Circular],
          ],
          "data": Object {
            "childs": Array [
              Object {
                "name": "Black",
              },
            ],
            "name": "Colour",
          },
          "depth": 0,
          "height": 1,
          "parent": null,
          "x": 40,
          "y": 10,
        },
        "x": 40,
        "y": 40,
      },
    ]
  }
  steps={20}
  width={200}
/>
`;
